### DISCUSSION TITLE
Requested Version "6.0.14" is not available for "Debian 12"! Available Versions: ">=7.0.3"

### DISCUSSION THREAD

**Discussion by k-gupta-2001** [typegoose/mongodb-memory-server#871] [Q&A] [answered]
## Versions

- NodeJS: 16.20.2
- mongodb-memory-server-*: 9.2.0 
- mongodb(the binary version): 0.0.0
- mongodb(the js package): 0.0.0
- mongoose: 8.3.4 
- system: debian 12 image (arm64 instance)

package: mongo-memory-server 

## What is your question?

When running my test cases , this is the error i am getting 

Requested Version "6.0.14" is not available for "Debian 12"! Available Versions: ">=7.0.3"
    Mongodb does not provide binaries for versions before 7.0.3 for Debian 12+ and also cannot be mapped to a previous Debian release

      14 |
      15 | async function start_database() {
    > 16 |     mongoServer = await MongoMemoryServer.create()
         |                   ^
      17 |     mongoUri = mongoServer.getUri()
      18 |     try {
      19 |         db = await mongoose.connect(mongoUri, {

      at MongoBinaryDownloadUrl.getDebianVersionString (node_modules/mongodb-memory-server-core/src/util/MongoBinaryDownloadUrl.ts:313:15)
      at start_database (__test__/integration/database/db.js:16:19)
      at Object.<anonymous> (__test__/setup.js:10:9)

Can you please provide steps to resolve this , i am stuck and not able to understand what to do

**Comment by hasezoey** [comment]
as the error says you will need to manually set a mongodb version to be used, the default for MMS 9.x is mongodb 6.0.x, see [Mongodb Server Versions](https://nodkz.github.io/mongodb-memory-server/docs/guides/mongodb-server-versions) on what the policy for mongodb versions is and what versions use what default version.

TL;DR: set [`MONGOMS_VERSION=7.0.3`](https://nodkz.github.io/mongodb-memory-server/docs/api/config-options#version) (or higher) as a environment variable or as a package.json config option

**Reply by hasezoey** [reply]
additionally, i just saw that you are using arm64, to my knowledge there are no debian arm builds. you can try the ubuntu arm builds through config option [`DISTRO`](https://nodkz.github.io/mongodb-memory-server/docs/api/config-options#distro)

**Reply by k-gupta-2001** [reply]
How to use distro , can you tell with an example , and where to configure it , i am unable to understand by the documentation 
`
mongoServer = await MongoMemoryServer.create({
        binary: {
            version: "7.0.3"
        }
    })
`
    
 This is what i have done , and now i am getting
 
  Download failed for url "https://fastdl.mongodb.org/linux/mongodb-linux-aarch64-debian12-7.0.3.tgz", Details:
    Status Code is 403 (MongoDB's 404)
    This means that the requested version-platform combination doesn't exist
    Try to use different version 'new MongoMemoryServer({ binary: { version: 'X.Y.Z' } })'
    List of available versions can be found here: https://www.mongodb.com/download-center/community/releases/archive

      at RedirectableRequest.<anonymous> (node_modules/mongodb-memory-server-core/src/util/MongoBinaryDownload.ts:383:17)

**Reply by hasezoey** [reply, marked as answer]
> Download failed for url "https://fastdl.mongodb.org/linux/mongodb-linux-aarch64-debian12-7.0.3.tgz", Details:

it is trying to find a `aarch64` (another arm64 name), which like i said in my last comment (and first comment to this thread), debian does not have arm builds, so you would need to try the ubuntu ones.

> How to use distro , can you tell with an example , and where to configure it , i am unable to understand by the documentation

currently config option `DISTRO` can only be configured via environment variables or through the package.json config option, not individually for a `.create` (or `new`).

i am not sure what about the documentation you dont understand, but here is a simple (hopefully complete) walkthrough about how to read the documentation and how to apply it:
Each config option in [Config Options](https://nodkz.github.io/mongodb-memory-server/docs/api/config-options#distro) has a name (the header) in this case `DISTRO`, which is the name it is referred to by me (and likely others), then it has a section outlining what the environment variable of that config option is (in this case `MONGOMS_DISTRO`) and the package.json config key (in this case `distro`).
At the end of the config options are the sections [How to use them in the package.json](https://nodkz.github.io/mongodb-memory-server/docs/api/config-options#how-to-use-them-in-the-packagejson) which outlines where and how to set the config options in the package.json (including subkey and name casing), and [How to use them as Environment Variables](https://nodkz.github.io/mongodb-memory-server/docs/api/config-options#how-to-use-them-as-environment-variables) in how to use those config options as environment variables (prefix and what values are accepted in boolean cases).

TL;DR: in your case try `MONGOMS_DISTRO=ubuntu-22.04`

**Reply by k-gupta-2001** [reply]
Thanks , that helps
